src/taipy/gui_core/GuiCoreLib.py

```diff
@@ -11,50 +11,108 @@
 
 import typing as t
 
-from taipy.core import Core, Scenario
+from dateutil import parser
+
+import taipy as tp
 from taipy.gui import Gui, State
 from taipy.gui.extension import Element, ElementLibrary, ElementProperty, PropertyType
 
 
 class GuiCoreContext:
+    __PROP_SCENARIO_CONFIG_ID = "config"
+    __PROP_SCENARIO_DATE = "date"
+    __PROP_SCENARIO_NAME = "name"
+    __SCENARIO_PROPS = (__PROP_SCENARIO_CONFIG_ID, __PROP_SCENARIO_DATE, __PROP_SCENARIO_NAME)
     _CORE_CHANGED_NAME = "core_changed"
-    _VAR_NAME = "__CCCtx"
+    _ERROR_VAR = "gui_core_error"
 
-    def __init__(self, gui: Gui, core: Core) -> None:
+    def __init__(self, gui: Gui) -> None:
         self.gui = gui
-        self.core = core
-        self.scenarios: t.List[Scenario]
+        self.scenarios: t.Optional[
+            t.List[t.Tuple[str, str, int, bool, t.Optional[t.List[t.Tuple[str, str, int, bool, None]]]]]
+        ] = None
+        self.scenario_configs: t.Optional[t.List[t.Tuple[str, str]]] = None
+
+    def __add_scenarios(self, res, scenarios):
```

Comment:
I suspect that this could be defined within the get_scenarios() method...

Reply:
It could but why?

Reply:
It could also be static

Reply:
The why is for easier-to-read code and scope concerns.
A function used only by another one has all the reasons to be located in the caller's scope.